sdks/java/core/src/main/java/org/apache/beam/sdk/values/PCollectionViews.java

```diff
@@ -528,5 +535,102 @@ public String toString() {
     public Map<TupleTag<?>, PValue> expand() {
       return Collections.<TupleTag<?>, PValue>singletonMap(tag, pCollection);
     }
+
+    @Override
+    public void finishSpecifyingOutput(
+        String transformName, PInput input, PTransform<?, ?> transform) {
+
+      Map<TupleTag<?>, PValue> inputValues = input.expand();
+      super.finishSpecifyingOutput(
+          // Make our name include both the input collection and how we're viewing it,
+          // for good error messages.
+          (inputValues.isEmpty()
+                  ? ""
+                  : (Iterables.getOnlyElement(inputValues.values()).getName() + "@"))
+              + transformName,
+          input,
+          transform);
+    }
+
+    private void writeObject(ObjectOutputStream os)
```

Comment:
Why do you need to implement Java's special serialization hook methods (writeObject/readObject)?